core/src/main/java/com/amplifyframework/core/model/AuthStrategy.java

```diff
@@ -25,23 +25,33 @@
      * Owner authorization specifies whether a user can access or operate against an object.  To use OWNER, the API
      * must have Cognito User Pool configured.
      */
-    OWNER,
+    OWNER("userPools"),
 
     /**
      * Group authorization specifies whether a group can access or operate against an object.  To use GROUPS, the API
      * must have Cognito User Pool configured.
      */
-    GROUPS,
+    GROUPS("userPools"),
 
     /**
      * The private authorization specifies that everyone will be allowed to access the API with a valid JWT token from
      * the configured Cognito User Pool. To use PRIVATE, the API must have Cognito User Pool configured.
      */
-    PRIVATE,
+    PRIVATE("userPools"),
 
     /**
      * The public authorization specifies that everyone will be allowed to access the API, behind the scenes the API
      * will be protected with an API Key. To use PUBLIC, the API must have API Key configured.
      */
-    PUBLIC,
+    PUBLIC("apiKey");
+
+    private final String defaultAuthProvider;
+
+    AuthStrategy(String defaultAuthProvider) {
+        this.defaultAuthProvider = defaultAuthProvider;
+    }
+
+    String getDefaultAuthProvider() {
+        return defaultAuthProvider;
+    }
```

Comment:
A switch statement would be appropriate here no? Using the default parameter for this doesn't sit right with me.

Reply:
Can you elaborate @raphkim?  What's your concern with the implementation?

Reply:
Not sure if this makes a difference here, but default in this context means the default auth provider for a given strategy. 

Also, I'm going to try to implement this as an enum, so it might look a little bit different.

Reply:
If you look in the `AuthorizationType.from(AuthRule authRuleAnnotation)` method, you'll see what my thinking was behind adding the default provider as a property of the AuthStrategy enum.